app/Http/Controllers/Service/Admin/TradersController.php

```diff
@@ -152,11 +153,27 @@ public function update(AdminNewUpdateTraderRequest $request, int $id)
                 // find the trader
                 $t = Trader::findOrFail($id);
 
-                // Update it
-                $t->fill([
+                // make an array of things to fill
+                $fillArray = [
                     'market_id' => $request->input('market'),
                     'name' => $request->input('name'),
-                ])->save();
+                ];
+
+                // If Input and Trader are in different states
+                if (boolval($request->input('disabled')) xor
+                    boolval($t->disabled_at)
+                ) {
+                    // ... then something's changed... set by input
+                    $fillArray['disabled_at'] = (boolVal($request->input('disabled')))
+                        // true-ish, make a date
+                        ? Carbon::now()
+                        // false-ish, set it null
+                        : null;
+                }
+                // or do nowt with owt
```

Comment:
should this comment be here?

Reply:
The xor is... tricky - I had to write a truth table on the whiteboard to work out the operations. 
Subsequently this section is perhaps over-commented, just in case
I'll make it less yorkshire though.